sklearn/linear_model/bayes.py

```diff
@@ -325,11 +330,15 @@ class ARDRegression(LinearModel, RegressorMixin):
     See examples/linear_model/plot_ard.py for an example.
     """
 
-    def __init__(self, n_iter=300, tol=1.e-3, alpha_1=1.e-6, alpha_2=1.e-6,
-                 lambda_1=1.e-6, lambda_2=1.e-6, compute_score=False,
-                 threshold_lambda=1.e+4, fit_intercept=True, normalize=False,
-                 copy_X=True, verbose=False):
-        self.n_iter = n_iter
+    @deprecated("Attribute n_iter was deprecated. Use 'max_iter' instead")
+    def __init__(self, n_iter=None, max_iter=300, tol=1.e-3, alpha_1=1.e-6,
+                 alpha_2=1.e-6, lambda_1=1.e-6, lambda_2=1.e-6,
+                 compute_score=False, threshold_lambda=1.e+4,
+                 fit_intercept=True, normalize=False, copy_X=True,
+                 verbose=False):
+        if n_iter is not None:
+            warnings.warn("'n_iter' was deprecated. Use 'max_iter' instead.")
```

Comment:
Please raise a `DeprecationWarning`
You also need to use this value if provided, so that the result is not modify by this PR:
```py
self.n_iter = n_iter
if n_iter is not None:
    warnings.warn(...)
    self.max_iter = n_iter
else:
    self.max_iter = max_iter
```
(same in other classes)